app/pages/settings/email.jsx

```diff
@@ -0,0 +1,299 @@
+import React from 'react';
+import apiClient from 'panoptes-client/lib/api-client';
+import talkClient from 'panoptes-client/lib/talk-client';
+import AutoSave from '../../components/auto-save';
+import handleInputChange from '../../lib/handle-input-change';
+
+class EmailSettingsPage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      page: 1,
+      projects: [],
+      projectPreferences: [],
+      talkPreferences: []
+    };
+    this.handlePreferenceChange = this.handlePreferenceChange.bind(this);
+    this.nameOfPreference = this.nameOfPreference.bind(this);
+    this.sortPreferences = this.sortPreferences.bind(this);
+    this.talkPreferenceOption = this.talkPreferenceOption.bind(this);
+    this.getProjectForPreferences(props.user);
+    this.getTalkPreferences();
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (!this.state.page === prevState.page) {
+      this.getProjectForPreferences();
+    }
+  }
+
+  getTalkPreferences() {
+    talkClient.type('subscription_preferences').get()
+    .then((preferences) => {
+      this.setState({
+        talkPreferences: preferences
+      });
+    });
+  }
+
+  getProjectForPreferences(user) {
+    return user.get('project_preferences', {
+      page: this.state.page,
+      sort: 'display_name'
+    })
+    .then((preferences) => {
+      if (preferences.length > 0) {
+        const projectIDs = preferences.map(pref => pref.links.project);
+        apiClient.type('projects').get({
+          id: projectIDs
+        })
+        .then((projects) => {
+          this.setState({
+            meta: preferences[0].getMeta(),
+            projectPreferences: preferences,
+            projects
+          });
+        });
+      }
+    })
+    .catch((error) => {
+      console.log('Something went wrong. Error: ', error);
+    });
+  }
+
+  handlePreferenceChange(preference, event) {
+    if (event.target.type === 'radio') {
```

Comment:
This is very fragile, in that it assumes that noone will ever add a radio button to this component in future. A better test would be to switch based on the target name, or to just have two event handler functions: one for the Talk preferences and another one for the project preferences.